controllers/provider-aws/pkg/infrastructure/actuator.go

```diff
@@ -0,0 +1,256 @@
+	a.config = config
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+	a.kubernetes = clientset
+	serverVersion, err := a.kubernetes.Discovery().ServerVersion()
+	if err != nil {
+		return err
+	}
+	a.serverVersion = serverVersion
+
+	return nil
+}
+
+func (a *actuator) InjectClient(client client.Client) error {
+	a.client = client
+	return nil
+}
+
+func (a *actuator) Exists(ctx context.Context, infrastructure *extensionsv1alpha1.Infrastructure) (bool, error) {
+	return infrastructure.Status.LastOperation != nil, nil
+}
+
+func (a *actuator) Create(ctx context.Context, config *extensionsv1alpha1.Infrastructure) error {
+	return a.reconcile(ctx, config)
+}
+
+func (a *actuator) Update(ctx context.Context, config *extensionsv1alpha1.Infrastructure) error {
+	return a.reconcile(ctx, config)
+}
+
+func (a *actuator) Delete(ctx context.Context, config *extensionsv1alpha1.Infrastructure) error {
+	return a.delete(ctx, config)
+}
+
+func (a *actuator) reconcile(ctx context.Context, infrastructure *extensionsv1alpha1.Infrastructure) error {
+	infrastructureConfig := &v1alpha1.InfrastructureConfig{}
+	_, _, err := a.decoder.Decode(infrastructure.Spec.ProviderConfig.Raw, nil, infrastructureConfig)
+	if err != nil {
+		a.injectStatusError(ctx, infrastructure, extensionsv1alpha1.LastOperationTypeReconcile, "could not decode providerConfig", 10, err)
+		return err
+	}
+
+	var (
+		namespace = infrastructure.Spec.SecretRef.Namespace
+		name      = infrastructure.Spec.SecretRef.Name
+	)
+
+	providerSecret := &corev1.Secret{}
+	if err := a.client.Get(ctx, kutil.Key(namespace, name), providerSecret); err != nil {
+		return err
+	}
+
+	chartRenderer, err := chartrenderer.New(a.kubernetes)
+	if err != nil {
+		a.injectStatusError(ctx, infrastructure, extensionsv1alpha1.LastOperationTypeReconcile, "could not create chart renderer", 30, err)
+		return err
+	}
+
+	chartValues, err := generateTerraformInfraConfigValues(infrastructure, infrastructureConfig, providerSecret)
+	if err != nil {
+		a.injectStatusError(ctx, infrastructure, extensionsv1alpha1.LastOperationTypeReconcile, "failed to generate terraform config values", 40, err)
+		return err
+	}
+
+	release, err := chartRenderer.Render(filepath.Join(aws.TerraformersChartsPath, "aws-infra"), "aws-infra", namespace, chartValues)
+	if err != nil {
+		a.injectStatusError(ctx, infrastructure, extensionsv1alpha1.LastOperationTypeReconcile, "could not render chart", 50, err)
+		return err
+	}
+
+	tf, err := a.getTerraformer(aws.TerrformerPurposeInfra, infrastructure.Namespace, infrastructure.Name)
+	if err != nil {
+		a.injectStatusError(ctx, infrastructure, extensionsv1alpha1.LastOperationTypeReconcile, "could not create terraformer object", 70, err)
+		return err
+	}
+
+	a.recorder.Event(infrastructure, corev1.EventTypeNormal, eventReasonCreation, "Applying Terraform config")
+	applyTerraform := func(secret *corev1.Secret, keysMap map[string]string) error {
```

Comment:
because sometimes its not clear what this code: 
 ```go 
tf.SetVariablesEnvironment(terraformer.GenerateVariablesEnvironment(providerSecret, accessKeysMap)).
			InitializeWith(terraformer.DefaultInitializer(a.client,
				release.FileContent("main.tf"),
				release.FileContent("variables.tf"),
				[]byte(release.FileContent("terraform.tfvars")))).
			Apply()
``` 

does immediately. By saying applyTerraform it gives context. And also because it was unreadable (your comment).

Reply:
IMHO it's pretty clear what happens (we finally call `Apply()` on the `tf` (=terraform) object. How does `applyTerraform` help then?
Anyway, generally use code comments if you want to explain what is happening, and use inline functions only if it comes with a programmatic benefit or similar